src/Symfony/Component/Notifier/Bridge/Mercure/MercureTransport.php

```diff
@@ -87,17 +86,15 @@ protected function doSend(MessageInterface $message): SentMessage
         ]), $options->isPrivate(), $options->getId(), $options->getType(), $options->getRetry());
 
         try {
-            $messageId = ($this->publisher)($update);
+            $messageId = $this->hub->publish($update);
 
             $sentMessage = new SentMessage($message, (string) $this);
             $sentMessage->setMessageId($messageId);
 
             return $sentMessage;
-        } catch (HttpExceptionInterface $e) {
-            throw new TransportException('Unable to post the Mercure message: '.$e->getResponse()->getContent(false), $e->getResponse(), $e->getCode(), $e);
-        } catch (ExceptionInterface $e) {
+        } catch (MercureRuntimeException $e) {
             throw new RuntimeException('Unable to post the Mercure message: '.$e->getMessage(), $e->getCode(), $e);
-        } catch (\InvalidArgumentException $e) {
+        } catch (MercureInvalidArgumentException $e) {
             throw new InvalidArgumentException('Unable to post the Mercure message: '.$e->getMessage(), $e->getCode(), $e);
```

Comment:
I suggest to throw a TransportException in both cases here.

Reply:
`Symfony\Component\Notifier\Exception\TransportException` to be constructed with a `ResponseInterface` but we haven't any response in these cases. What do you you suggest?

Reply:
Maybe a simple `Symfony\Component\Notifier\Exception\RuntimeException` will do the work?

Reply:
> Maybe a simple Symfony\Component\Notifier\Exception\RuntimeException will do the work?

👍 from my side

WDYT @chalasr ?

Reply:
👍 Right